www/main/navigator/misc/components/favoritePane.jsx

```diff
@@ -0,0 +1,180 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import isOnline from 'is-online';
+import { useStoreActions, useStoreState } from 'easy-peasy';
+
+import { fetchFavShabad, removeFromFav } from '../utils';
+import banidb from '../../../banidb';
+
+const remote = require('@electron/remote');
+
+const { i18n } = remote.require('./app');
+
+export const FavoritePane = ({ className }) => {
+  const {
+    activeShabadId,
+    initialVerseId,
+    versesRead,
+    isCeremonyBani,
+    isSundarGutkaBani,
+    homeVerse,
+    activeVerseId,
+    singleDisplayActiveTab,
+    favShabad,
+  } = useStoreState((state) => state.navigator);
+  const {
+    setActiveShabadId,
+    setInitialVerseId,
+    setVersesRead,
+    setIsCeremonyBani,
+    setIsSundarGutkaBani,
+    setHomeVerse,
+    setActiveVerseId,
+    setSingleDisplayActiveTab,
+    setFavShabad,
+  } = useStoreActions((state) => state.navigator);
+
+  const { userToken } = useStoreState((state) => state.app);
+  const [parsedFav, setParsedFav] = useState([]);
+  const [isFetching, setFetching] = useState(false);
+  const [errorMessage, setError] = useState('');
+
+  const initialFetch = async () => {
+    const onlineValue = await isOnline();
+    if (onlineValue) {
+      if (userToken) {
+        setError('');
+        setFetching(true);
+        const data = await fetchFavShabad(userToken);
+        setFetching(false);
+        setFavShabad([...data]);
+      } else {
+        setError(i18n.t('FAV_SHABAD.LOGGED_OUT'));
+        setFavShabad([]);
+      }
```

Comment:
I feel there should be valid exception handling instead of if else

Reply:
Added try catch block inside the if statement. We need if else for the logout message.